keps/sig-cli/2906-kustomize-function-catalog/README.md

```diff
@@ -569,11 +570,235 @@ This proposal introduces extension capabilities to Kustomize that may expose use
+                type: string
+            runtime:
+              description: |
+                The runtime information about the KRM function. At least one of 
+                container and exec must be set.
+              type: object
+              properties:
+                container:
+                  description: The runtime information for container-based KRM function.
+                  type: object
+                  required:
+                    - image
+                  properties:
+                    image:
+                      description: The image name of the KRM function.
+                      type: string
+                    sha256:
+                      description: |
+                        The digest of the image that can be verified against. It
+                        is required only when the image is using semver.
+                      type: string
+                    requireNetwork:
+                      description: If network is required to run this function.
+                      type: boolean
+                    requireStorageMount:
+                      description: If storage mount is required to run this function.
+                      type: boolean
+                exec:
+                  description: The runtime information for exec-based KRM function.
+                  type: object
+                  required:
+                    - platform
+                  properties:
+                    platforms:
+                      description: Per platform runtime information.
+                      type: array
+                      items:
+                        type: object
+                        required:
+                          - bin
+                          - os
+                          - arch
+                          - uri
+                          - sha256
+                        properties:
+                          bin:
+                            description: The binary name.
+                            type: string
+                          os:
+                            description: The target operation system to run the KRM function.
+                            type: string
+                            enum:
+                              - linux
+                              - darwin
+                              - windows
+                          arch:
+                            description: The target archtechture to run the KRM function.
+                            type: string
+                            enum:
+                              - amd64
+                              - arm64
+                          uri:
+                            description: The location to download the binary.
+                            type: string
+                          sha256:
+                            description: The degist of the binary that can be used to verify the binary.
+                            type: string
+      home:
```

Comment:
Do we need this, usage _and_ examples? Do you think they'll often point different places, and if so, why?

Reply:
IMO `home` is really optional, while `usage` and `examples` are required.
`usage` should be the man page for the function. A pointer to a markdown file should be good.
In kpt project, each function example is self-contained directory. It has some KRM resources as input, fn config and some configuration to test the example. We can ensure the examples always work. IMO we can do something similar here.
WDYT?

Reply:
Sounds reasonable. Let's start with this, and we can evolve it if needed.